File: UILabel/UILabel+Extension.m
//
//  UILabel+Extension.m
//  loginView
//
//  Created by Yakir on 16/4/8.
//  Copyright © 2016年 PAJK. All rights reserved.
//

#import "UILabel+Extension.h"

@implementation UILabel (Extension)

+ (instancetype)ylLabelWithTitle:(NSString *)title
                        fontSize:(CGFloat)fontSize
                           frame:(CGRect)frame
                 textColor:(UIColor *)color {
    
    UILabel *label = [UILabel new];
    label.frame = frame;
    label.text = title;
    label.font = [UIFont systemFontOfSize:fontSize];
    label.textAlignment = NSTextAlignmentCenter;
    label.textColor = color;
    
    return label;
}

+ (instancetype)ylLabelWithTitle:(NSString *)title fontSize:(CGFloat)fontSize {
    return [self ylLabelWithTitle:title fontSize:fontSize frame:CGRectZero textColor:[UIColor blackColor]];
}

+ (instancetype)ylLabelWithTitle:(NSString *)title
                        fontSize:(CGFloat)fontSize
                       textColor:(UIColor *)color {
    return [self ylLabelWithTitle:title fontSize:fontSize frame:CGRectZero textColor:color];
}

@end

File: Swift/UILabel+Extension.swift
//
//  UILabel+Extension.swift
//  SmallVillage
//
//  Created by 李 on 16/7/27.
//  Copyright © 2016年 chenglv. All rights reserved.
//

import UIKit

extension UILabel {
    
    /// 创建 UILabel
    ///
    /// - parameter text:      text
    /// - parameter fontSize:  fontSize，默认 15
    /// - parameter color:     color，默认 黑色
    /// - parameter alignment: alignment，默认左对齐
    ///
    /// - returns: UILabel
    convenience init(yl_text text: String,
                             fontSize: CGFloat = 15,
                             color: UIColor = UIColor.blackColor(),
                             alignment: NSTextAlignment = .Left) {
        
        self.init()
        
        self.text = text
        self.textColor = color
        self.font = UIFont.systemFontOfSize(fontSize)
        self.textAlignment = alignment
        
        self.numberOfLines = 0
        
        // 自动调整大小
        sizeToFit()
    }
}
